### PR TITLE
Fix compatibility problems with vulkan 1.0 devices

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**

### CODE REVIEW

**Comment by jeremyg-lunarg** [comment]
Passes local testing on linux and android (pixel 3a). Internal CI for windows in progress

File: layers/synchronization2.cpp [hunk 1]
```diff
@@ -397,6 +401,10 @@ DeviceFeatures::DeviceFeatures(uint32_t api_version, const VkDeviceCreateInfo* c
     if (api_version >= VK_MAKE_VERSION(1, 2, 0)) {
         timelineSemaphore = true;
     }
+    if (create_info->pEnabledFeatures != nullptr) {
```
Comment (Tobski) [inline layers/synchronization2.cpp:404]: Is this guaranteed to be nullptr if there's a GPDP2 structure in pNext?

Comment (jeremyg-lunarg) [reply layers/synchronization2.cpp:404]: VUID-VkDeviceCreateInfo-pNext-00373 
If the <code>pNext</code> chain includes a <a href=\"#VkPhysicalDeviceFeatures2\">VkPhysicalDeviceFeatures2</a> structure, then <code>pEnabled  Features</code> must be <code>NULL</code>

File: layers/synchronization2.cpp [hunk 2]
```diff
@@ -1318,6 +1326,11 @@ VKAPI_ATTR VkResult VKAPI_CALL CreateRenderPass2(VkDevice device, const VkRender
                            VkRenderPass* pRenderPass) {
     auto device_data = GetDeviceData(device);
 
+    if (device_data->vtable.CreateRenderPass2 == nullptr) {
+        std::cerr << __func__ << " Device does not support CreateRenderPass2" << std::endl;
```
Comment (sfricke-samsung) [inline layers/synchronization2.cpp:1330]: Can we use the `LOG()` macro, mainly for Android Logcat printing (feel free to spice up the non-Android variation I quick threw there)

Comment (jeremyg-lunarg) [reply layers/synchronization2.cpp:1330]: good idea

**Comment by jeremyg-lunarg** [comment]
Internal Win32 CI passes (job 52), local testing with linux and Android (pixel 3a / vulkan 1.1) passes as well.

**Comment by jeremyg-lunarg** [comment]
dEQP-VK.synchronization2.* passes on linux

**Pull request merged**